Sources/Customization/Protocols/OPTEventProcessor.swift

```diff
@@ -0,0 +1,34 @@
+/****************************************************************************
+* Copyright 2019, Optimizely, Inc. and contributors                        *
+*                                                                          *
+* Licensed under the Apache License, Version 2.0 (the "License");          *
+* you may not use this file except in compliance with the License.         *
+* You may obtain a copy of the License at                                  *
+*                                                                          *
+*    http://www.apache.org/licenses/LICENSE-2.0                            *
+*                                                                          *
+* Unless required by applicable law or agreed to in writing, software      *
+* distributed under the License is distributed on an "AS IS" BASIS,        *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+* See the License for the specific language governing permissions and      *
+* limitations under the License.                                           *
+***************************************************************************/
+    
+
+import Foundation
+
+public protocol OPTEventProcessor {
+    
+    func process(event: UserEvent, completionHandler: DispatchCompletionHandler?)
```

Comment:
Create a ProcessorCompletionHandler (it can just take the user event and that way you can do the json encoding async in another thread.